src/global/reco/ParticleFlow_factory.cc

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) Derek Anderson
+
+#include <memory>
+#include <exception>
+#include <stdexcept>
+#include <JANA/JEvent.h>
+#include <JANA/JException.h>
+// event data model definitions
+#include <edm4eic/ClusterCollection.h>
+#include <edm4eic/ReconstructedParticleCollection.h>
+// for geometry services
+#include <DD4hep/Detector.h>
+// factory-specific includes
+#include "ParticleFlow_factory.h"
+
+namespace eicrecon {
+
+  void ParticleFlow_factory::Init() {
+
+    auto app = GetApplication();
+
+    // This prefix will be used for parameters
+    std::string plugin_name = GetPluginName();
+    std::string param_prefix = plugin_name + ":" + GetTag();
+
+    // SpdlogMixin logger initialization, sets m_log
+    InitLogger(GetApplication(), GetPrefix(), "info");
+
+    // Algorithm configuration
+    auto cfg = GetDefaultConfig();
+
+    app->SetDefaultParameter(param_prefix + ":flowAlgo",      cfg.flowAlgo);
+    app->SetDefaultParameter(param_prefix + ":ecalSumRadius", cfg.ecalSumRadius);
+    app->SetDefaultParameter(param_prefix + ":hcalSumRadius", cfg.hcalSumRadius);
+    app->SetDefaultParameter(param_prefix + ":ecalFracSub",   cfg.ecalFracSub);
+    app->SetDefaultParameter(param_prefix + ":hcalFracSub",   cfg.hcalFracSub);
+
+    // initialize particle flow algorithm
+    m_pf_algo.applyConfig(cfg);
+    m_pf_algo.init(logger());
+
+    // initialize geometry service
+    m_geoSvc = app -> template GetService<DD4hep_service>();
+
+  }  // end 'Init()'
+
+
+
+  void ParticleFlow_factory::BeginRun(const std::shared_ptr<const JEvent> &event) {
+
+    // Nothing to do here
+
+  }  // end 'BeginRun(std::shared_ptr<JEvent&>)'
+
+
+
+  void ParticleFlow_factory::Process(const std::shared_ptr<const JEvent> &event) {
+
+    // for calo inputs
+    std::vector<const edm4eic::ClusterCollection*> vecECalInput(m_const.nCaloPairs);
+    std::vector<const edm4eic::ClusterCollection*> vecHCalInput(m_const.nCaloPairs);
+
+    // grab and organize input collections
+    TrkInput trkInput;
+    for (const std::string& input_tag : GetInputTags()) {
```

Comment:
The looping over input tags gives you flexibility but that's not what we need here. I think you should enforce the interface: first input = ReconstructedParticleCollection, second input = TrackSegmentCollection, third input = ClusterCollection, etc.